src/sidebar/components/moderation-banner.js

```diff
@@ -1,67 +1,108 @@
 'use strict';
 
-const annotationMetadata = require('../annotation-metadata');
-
-// @ngInject
-function ModerationBannerController(store, flash, api) {
-  const self = this;
+const { createElement } = require('preact');
+const classnames = require('classnames');
+const propTypes = require('prop-types');
 
-  this.flagCount = function() {
-    return annotationMetadata.flagCount(self.annotation);
-  };
+const annotationMetadata = require('../annotation-metadata');
+const useStore = require('../store/use-store');
+const { withServices } = require('../util/service-context');
 
-  this.isHidden = function() {
-    return self.annotation.hidden;
-  };
+/**
+ * Banner allows moderators to hide/unhide the flagged
+ * annotation from other users.
+ */
+function ModerationBanner({ annotation, api, flash }) {
+  // actions
+  const store = useStore(store => ({
+    hide: store.hideAnnotation,
+    unhide: store.unhideAnnotation,
+  }));
 
-  this.isHiddenOrFlagged = function() {
-    const flagCount = self.flagCount();
-    return flagCount !== null && (flagCount > 0 || self.isHidden());
-  };
+  const flagCount = annotationMetadata.flagCount(annotation);
 
-  this.isReply = function() {
-    return annotationMetadata.isReply(self.annotation);
+  const isHiddenOrFlagged = function() {
```

Comment:
@robertknight pointed out to me that this function is just a single `return` statement and could be assigned a value instead, e.g.:

`const isHiddenOrFlagged = flagCount !== null && (flagCount > 0 || annotation.hidden);`